doc/source/msg.rst

```diff
@@ -186,16 +186,20 @@ of immediate motion.
 
 stage
 +++++
+Instruct the RunEngine to stage the object. This calls ``obj.stage()``.
 
 
 trigger
 +++++++
 
-This will call the ``obj.trigger`` method and cache the returned status object.
+This will call the ``obj.trigger`` method and cache the returned status object
+and caches the returned status object.
 
 
 unstage
 +++++++
+Instruct the RunEngine to unstage the object. This calls ``obj.unstage()`` and
```

Comment:
Stage and unstage return a list of the Device(s) that were (un)staged, not status objects.

Why not status objects? We debated this one for awhile. Operations performed during staging are supposed to involve twiddling configuration, and should happen fast. Staging should not involve lengthy `set` calls.

Why a list of the objects staged? Staging a Device causes that Device's component Devices (if any) to also be staged. All of these children are added to a list, along with `[self]`, and returned by `Device.stage()`, so that the _plan_ can keep track of what has been staged, like so:

```python
devices_staged = yield Msg('stage', device)
```

Why would the plan want to know that? It needs to avoid accidentally trying to stage something twice, such as a staging a parent and then trying to also stage its child. It's important to avoid that because staging something redundantly raises an error.

Reply:
thanks @danielballan ! i ll add most of your wording to the PR.